### DISCUSSION TITLE
Module 1.1 - Trouble running main.py

### DISCUSSION THREAD

**Discussion by mn9102** [AccelerationConsortium/ac-microcourses#20] [Hello World] [answered]
Hi, I get the following import error when I run main.py on the microcontroller, and it seems to start disconnecting every few minutes and the light at GP15 flashes before disconnecting. Thank you!

![Screenshot 2024-07-04 223838 Error Msg - edited](https://github.com/AccelerationConsortium/ac-microcourses/assets/144367290/a42e5426-e9c6-47cf-9bc6-e6a3c7e23899)

![20240704_221748](https://github.com/AccelerationConsortium/ac-microcourses/assets/144367290/ebe8aeb9-8941-4e32-bb15-6461da57f18d)

**Comment by sgbaird** [comment]
Hi @mn9102, the issue with `ussl` is related to versioning. Which MicroPython version is installed? Try using the latest SDL Demo package: https://github.com/sparks-baird/self-driving-lab-demo/releases/latest.

**Reply by mn9102** [reply]
I have the `v1.24.0-preview.100.g358e501e7` version installed.

**Reply by sgbaird** [reply]
Maybe try `v1.23`. Sometimes the preview builds can have unexpected issues. Also, try using `v0.8.9` for SDL Demo.

**Reply by mn9102** [reply]
I tried running the updated Demo package with `v1.23`, unfortunately, it is still showing the same import error.

**Reply by sgbaird** [reply]
@mn9102 if you're using the most recent version of the SDL Demo package, you shouldn't be getting a ussl error (https://github.com/sparks-baird/self-driving-lab-demo/blob/main/src/public_mqtt_sdl_demo/lib/umqtt/simple.py). Did you already upload the `v0.8.9` files to your microcontroller? This should also include the `lib` folder.

**Reply by mn9102** [reply]
Got it, the files didn't seem to override properly when I uploaded it earlier, but with the updated files it gives me `MQTTException: 5`, and then it seems like the microcontroller disconnects. 

![image](https://github.com/AccelerationConsortium/ac-microcourses/assets/144367290/3db8faa8-90aa-4e7f-8aea-7880c6a375fe)

**Reply by sgbaird** [reply, marked as answer]
Are you using your own HiveMQ broker or the default one that was supplied? (I suggest trying with the default one if you haven't already)

**Reply by mn9102** [reply]
Thank you very much! It is working now with the default HiveMQ broker.

**Reply by sgbaird** [reply]
Nice! @mn9102 for the assignment, I believe you will need to use your own HiveMQ broker. Perhaps there is an issue with the setup of the custom one. You can do some debugging using your HiveMQ Cloud Console (mentioned in the assignment README I think). You can get to it by clicking the "web client" navigation tab:

![image](https://github.com/AccelerationConsortium/ac-microcourses/assets/45469701/f47b4d52-0917-47b6-b80b-d946cd28a659)
